multi-agent/lib/mas/core/runtime_binder.py

```
@@ -0,0 +1,153 @@
+    """One bindable runtime capability.
+
+    Self-contained: knows which protocol to check, how to apply a
+    value, and how to clean up.  The binder iterates slots without
+    any ``if/elif`` chains.
+    """
+
+    protocol: type
+    apply_fn: Callable[[Any, T], None]
+    remove_fn: Callable[[Any], None]
+
+    def supports(self, node: Any) -> bool:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `node`

(ANN401)